drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -610,49 +610,55 @@
+          Important note: if all the inserted IOAM-Data-Fields are supposed to be
+          modified by other IOAM nodes on the path, or if there is no IOAM-Data-Field
+          inserted at all, then the encapsulating node MUST NOT use an Integrity
           Protected Option-Type.</t>
 
           <t>A transit node generates a signature over the Signature field and
           the hash of IOAM-Data-Fields it has inserted, i.e.,
-          sign(Signature || hash(IOAM-Data-Fields)). IOAM-Data-Fields that are updated
-          in-place MUST be excluded from the signature (e.g., the POT Cumulative
-          field). The signature is stored in the Signature field of the
-          Integrity Protection subheader. Important note: if all the IOAM-Data-Fields
-          involved are updated in-place, or if there is no IOAM-Data-Field involved,
-          the transit node MUST NOT generate a signature and MUST NOT update the
-          Signature field.</t>
-
-          <t>The decapsulating node behaves exactly the same as a transit node.
-          The only difference is that the signature MAY NOT be required when the
-          decapsulating node is also the Validator, for obvious performance
-          reasons.</t>
-
-          <t>The Validator recreates the signature over IOAM-Data-Fields collected
-          and checks the integrity against the Signature field. In order to
-          recompute the signature, the Validator iteratively follows the same
-          procedure as for the encapsulating, transit and decapsulating nodes,
-          in that order. It is trivial in some cases (e.g., POT Type-0 or E2E),
-          where only the encapsulating node generates a signature. For other
-          cases where transit nodes also generate a signature (e.g., Trace
-          Option-Types), node-ids MUST be recorded. Details on how the mapping
-          between node-ids and keys is implemented are outside the scope of this
-          document.</t>
+          sign(Signature || hash(IOAM-Data-Fields)). IOAM-Data-Fields modified
+          in-place by the transit node MUST be excluded from the signature
+          (e.g., the POT Cumulative field). The signature is stored in the Signature
+          field of the Integrity Protection subheader. Important note: if the
+          transit node does not insert IOAM-Data-Fields (e.g., it only modifies
+          IOAM-Data-Fields in-place, or does nothing), then the transit node
+          MUST NOT generate a signature and MUST NOT update the Signature field.</t>
+
+          <t>The decapsulating node (aka the Validator) is responsible for the
+          integrity verification of the IOAM-Data-Fields collected. For obvious
+          performance reasons and because it is the Validator, the decapsulating
+          node MUST NOT generate a signature based on IOAM-Data-Fields it has
+          inserted, if any, and therefore MUST NOT update the Signature field.
```

Comment:
I would suggest to avoid terms like "for obvious performance reasons". What might obvious today, might be really strange in 10 years time. IMHO we can exclude the note on performance reasons and simply say:

"The decapsulating node, serving as the Validator, MUST NOT generate a signature based on ..."

Reply:
Solved by commit [968f166](https://github.com/inband-oam/ietf/pull/252/commits/968f1662ce30060bfec5035949832775d38c9fef) in this PR